docs/map.rst

```diff
@@ -0,0 +1,602 @@
+off the ``Map.`` qualifier you'll get confusing type errors.
+
+::
+
+    import Prelude hiding (lookup)
+
+    import Data.Map.Strict (Map)
+    import qualified Data.Map.Strict as Map
+
+    import Data.IntMap (IntMap)
+    import qualified Data.IntMap.Strict as IntMap
+
+
+Common API Functions
+--------------------
+
+.. TIP::
+   All of these functions that work for ``Map`` will also work for ``IntMap``,
+   which has the key type ``k`` specialized to ``Int``. Anywhere that you
+   see ``Map Int v`` you can replace it with ``IntMap v``.
+
+.. NOTE::
+   ``fromList [some,map,entries]`` is how a ``Map`` is printed.
+
+
+Construction and Conversion
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Create an empty map
+"""""""""""""""""""
+
+::
+
+    Map.empty :: Map k v
+    Map.empty = ...
+
+:haddock_short:`Data.Map.Strict#empty` creates a map without any entries.
+
+::
+
+    Map.empty
+    > fromList []
+
+Create a map with one entry (singleton)
+"""""""""""""""""""""""""""""""""""""""
+
+::
+
+    Map.singleton :: k -> v -> Map k v
+    Map.singleton key value = ...
+
+:haddock_short:`Data.Map.Strict#singleton` creates a map with a single
+``(key,value)`` entry in it.
+
+::
+
+    Map.singleton 1 "one"
+    > fromList [(1,"one")]
+
+    Map.singleton "containers" ["base"]
+    > fromList [("containers",["base"])]
+
+Create a map from a list
+""""""""""""""""""""""""
+
+::
+
+    Map.fromList :: Ord k => [(k, v)] -> Map k v
+    Map.fromList xs = ...
+
+:haddock_short:`Data.Map.Strict#fromList` creates a map containing the entries
+of the list ``xs`` where the keys comes from the first entries of the pairs and
+the values from the second. If the same key appears more than once then the last
+value is taken.
+
+::
+
+    Map.fromList []
+    > fromList []
+
+    Map.fromList [(1,"uno"), (1,"one"), (2,"two"), (3,"three")]
+    > fromList [(1,"one"),(2,"two"),(3,"three")]
+
+There's another incredibly useful function for constructing a map from a list::
+
+    Map.fromListWith :: Ord k => (a -> a -> a) -> [(k, a)] -> Map.Map k a
+    Map.fromListWith f xs = ...
+
+:haddock_short:`Data.Map.Strict#fromListWith` allows you to build a map from a
+list ``xs`` with repeated keys, where ``f`` is used to "combine" (or "choose")
+values with the same key.
+
+::
+
+    -- Build a map from a list, but only keep the largest value for each key.
+    Map.fromListWith max [("a", 2), ("a", 1), ("b", 2)]
+    > fromList [("a",2),("b",2)]
+
+    -- Build a histogram from a list of elements.
+    Map.fromListWith (+) (map (\x -> (x, 1)) ["a", "a", "b", "c", "c", "c"])
+    > fromList [("a",2),("b",1),("c",3)]
+
+    -- Build a map from a list, combining the string values for the same key.
+    Map.fromListWith (++) [(1, "a"), (1, "b"), (2, "x"), (2, "y")]
+    > fromList [(1,"ba"),(2,"yx")]
+
+
+
+Create a list from a map
+""""""""""""""""""""""""
+
+::
+
+    Map.elems :: Map k v -> [v]
+    Map.elems m = ...
```

Comment:
This function isn't important enough, IMO.

Reply:
Removed.